### PR TITLE
Fix regression from fixing relative key handling where custom field no longer exists in the database

### PR DESCRIPTION
**Pull request by seamuslee001**
… to load if a custom field has been deleted which was stored in the form values array

Overview
----------------------------------------
@eileenmcnaughton this is the regression i have found, we will need to backport the fix to 5.19 (fix incoming)

### CODE REVIEW

File: tests/phpunit/CRM/Contact/BAO/SavedSearchTest.php [hunk 1]
```diff
@@ -96,6 +96,9 @@ public function testDefaultValues() {
     $defaults = $sg->setDefaultValues();
 
     $this->checkArrayEquals($defaults, $formValues);
+    $this->callAPISuccess('CustomField', 'delete', ['id' => $this->ids['CustomField']['int']]);
+    $defaults = $sg->setDefaultValues();
+    print_r($defaults);
```
Comment (eileenmcnaughton) [inline tests/phpunit/CRM/Contact/BAO/SavedSearchTest.php]: debug

**Comment by seamuslee001** [comment]
@eileenmcnaughton have just pushed the fix

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I'm mostly OK with this but I do wonder what should really happen to custom groups when a custom field within them is disabled. This feels like the group will be missing a filter & 'no-one will know'

**Comment by seamuslee001** [comment]
@eileenmcnaughton yeh i dunno however this does get past a fatal for me. I feel like the advanced search quietly drops things. We could try and be more noisey but maybe something more for master/5.21

**Comment by seamuslee001** [comment]
@eileenmcnaughton have created the 5.19 port here https://github.com/civicrm/civicrm-core/pull/16003

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I guess I'm not 100% sure if the fatal isn't more appropriate as you have a group that is not doing what it was created to do (ie filter on a particular value)  & hence if you mail to it you might be mailing the wrong people. 

I'm OK to merge this to the PR though as it returns to previous behaviour but....

**Pull request merged**